Frameworks/CoreGraphics/DWriteWrapper.mm

```diff
@@ -14,184 +14,101 @@
 //
 //******************************************************************************
 
-// #1207: Do not move this block, it has to come first for some reason
 #include <COMIncludes.h>
 #import <wrl/implements.h>
 #include <COMIncludes_End.h>
 
 #import <CoreGraphics/DWriteWrapper.h>
+#import <StringHelpers.h>
 
-#import <unordered_map>
-#import <vector>
 #import <functional>
-#import <mutex>
+#import <vector>
 #import <type_traits>
 
-using namespace std;
+#import "DWriteFontBinaryDataLoader.h"
+#import "DWriteFontCollectionHelper.h"
+#import "DWriteFontBinaryDataCollectionLoader.h"
+
 using namespace Microsoft::WRL;
 
 static const wchar_t* TAG = L"_DWriteWrapper";
 static const wchar_t* c_defaultUserLanguage = L"en-us";
 
-// Static collection which holds all fonts registered by user programatically, through plist, etc.
-static ComPtr<IDWriteFontCollection> _userCreatedFontCollection;
-static std::mutex _userCreatedFontCollectionMutex;
-
 /**
- * Private helper method to return the user set default locale string.
- *
- * @return use set locale string as wstring.
+ * Private concrete implementation of DWriteFontCollectionHelper.
+ * Wraps around the system font collection.
+ * Intended as a singleton.
  */
-static wstring __GetUserDefaultLocaleName() {
-    wchar_t localeName[LOCALE_NAME_MAX_LENGTH];
-    int defaultLocaleSuccess = GetUserDefaultLocaleName(localeName, LOCALE_NAME_MAX_LENGTH);
+class SystemFontCollectionHelper : public DWriteFontCollectionHelper {
+protected:
+    ComPtr<IDWriteFontCollection> _GetFontCollection() {
+        // Seems okay to fail-fast in this case - hard to continue without any system font collection
+        ComPtr<IDWriteFactory> dwriteFactory;
+        THROW_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
 
-    // If the default locale is returned, find that locale name, otherwise use "en-us".
-    return wstring(defaultLocaleSuccess ? localeName : c_defaultUserLanguage);
-}
+        ComPtr<IDWriteFontCollection> fontCollection;
+        THROW_IF_FAILED(dwriteFactory->GetSystemFontCollection(&fontCollection));
```

Comment:
nit: Would prefer to keep this static to only pull once rather than getting every time this is called